### DISCUSSION TITLE
Announcing RenderCV Web App and REST API

### DISCUSSION THREAD

**Discussion by sinaatalay** [rendercv/rendercv-app#1] [Announcements]
I created RenderCV because I needed a tool to help create my LaTeX CVs. I wanted to focus on my content rather than getting lost in the details of LaTeX or sometimes focus solely on LaTeX without being distracted by content.

That's RenderCV: It takes your content and creates the LaTeX file using the recipe you provided.

Then, the obvious next thing that comes to mind: **What if RenderCV was a cross-platform app (iOS, Android, and Web) so that I could manage, control, and store my CVs from my phone and the web?**

Imagine an app:

-   When someone requests your CV, you can simply open your phone and click the share button.
-   You can update your CV from anywhere in the world using your phone or computer.
-   You can version-control your CVs, like storing a history: Which CV did you use to apply for PhDs 10 years ago? It will be stored and locked securely.
-   You have complete control over the LaTeX code, just like using RenderCV.
-   Your CVs will always be reproducible because they are written with RenderCV, an open-source platform.
-   And many more possibilities?

I believe RenderCV lays a great foundation for such an app. And I believe there is a need for an app like this. Therefore, a friend of mine and I have been working on this project, and today, we are launching the beta version, which is currently available on the web only.

We aim to complete this app for all platforms as quickly as possible, and we'd love to hear your feedback and comments as we move forward. Your input is highly valued.

[rendercv.com](https://rendercv.com)
[REST API](https://api.rendercv.com/docs)

**Comment by mattmaniak** [comment]
Just started using RenderCV locally and I am happy to hear it's growing! A thing that blocked me a little are buttons on the main page. Only text is treated as a link so clicking a button outside a text does nothing. I think that making full buttons clickable will improve navigation and UX. Thanks.
![image](https://github.com/user-attachments/assets/89252f61-da65-4f71-95a7-d0e180e390fe)

**Reply by sinaatalay** [reply]
Thank you, that's a very good point. We have just updated it.

**Reply by mattmaniak** [reply]
Update like a rocket! Works as I imagined now. Thank you.

**Comment by xuganyu96** [comment]
Already compiling the LateX locally, but this is awesome. Will be using this resume template for the upcoming job search. Thank you so much!

**Comment by Sha-yol** [comment]
Awesome app! Two comments:
1. Dark mode feature is cool, but doesn't persist when refreshing the page.
2. Would be cool to have an option to download the html/markdown, to put in a spell checker, as when compiling locally.
Thanks again for the good work!

**Reply by sinaatalay** [reply]
Thank you for the feedback!

**Comment by jaepil-choi** [comment]
Hey you know what? I was also frustrated with absurd Word resume nonsense one night and suddenly came up with LaTeX resume app idea. Asked GPT for demo code and stuff but then I thought: 'Hmm, someone on the other side of the globe might already have come up with the same idea'. 
Here it is. Something I was looking for. Now I don't have to re-invent the wheel myself. Super great idea. You just saved many uni students purchasing overpriced mediocre resume tools online...

**Reply by sinaatalay** [reply]
Thank you!

**Comment by redfyve** [comment]
I first want to say, I love the web application. It's going to come in handy especially for me to apply to internships.
Wondering if there could be a new entry type added that allows for multiple roles under the same company, or in my case multiple roles under the same technical team. Thanks!

**Comment by daboone** [comment]
Hi,

I  love the render CV I have it running locally before that though I was trying the online one and lost all the content I created when I signed in that was irritating. 
Also returning to it there is a spinner and if you fire up Debug tools in Chrome you will see an error loading url: https://app.rendercv.com/flutter.js.map: Line 336 Column 22. In the flutter.js file.. I suspect it's this reason the spinner doesn't stop loading.

Also Chrome is warning about an uncaught promise.
flutter_bootstrap.js:16 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'style')
    at v.onEntrypointLoaded [as _onEntrypointLoaded] (flutter_bootstrap.js:16:38)

Hope you can sort it out it's probably spoiling some people's introduction to it.

Ohh one final thing is there anyway to encourage page breaks at specific points in the yaml? 

Best of luck.

**Comment by rahulw1981** [comment]
I have encountered countless frustrating issues with the resume rendering on this platform. Every single time I switch from one CV file to another, the elements either disappear or don't load at all, completely breaking the document. This has not only wasted an enormous amount of my time but has also made the entire process stressful and downright unbearable. Honestly, this has been the worst experience I've ever had with any resume editing tool. It is unacceptable for a platform claiming to support professional work to have such unreliable rendering. Please fix this immediately and implement a real-time rendering button so users can preview their resumes accurately and without constant glitches. The current workflow is inefficient, clunky, and extremely user-unfriendly. This needs urgent attention.

**Reply by sinaatalay** [reply]
I understand your frustration, but please remember that RenderCV is a free and open-source hobby project, built and maintained in our spare time. We’ve shared it with the community as-is, without any guarantees or commercial support.

Unfortunately, we haven’t had much time to work on it recently, and some dependency updates have broken key functionality.

We hope it becomes useful to you again once things are fixed. Until then, thanks for your patience and understanding.

**Comment by v1rtu0z** [comment]
First of all, this is an amazing project! You might want to add a ko-fi button to the tool itself, I bet many folks would be happy to click it once it helps them find a job ;)

**Reply by sinaatalay** [reply]
Thanks so much! Sorry for the super late response, we've just added the Ko-fi button to the web app. Really appreciate the suggestion and your support! :)

**Comment by jaepil-choi** [comment]
I think we need more default templates as it is not easy to create latex or typst based resume for a normal person. Also, it would be nice to have a template that has a image section that you can put your profile image or QR code for linktree in it.

**Comment by DrAndiLowe** [comment]
How do I use LaTeX in rendercv YAML? For example, how would I write the following in my CV?
```
Investigated the search potential for the Higgs boson using the $H \rightarrow b\bar{b}$ decay channel
```

I'm unable to get the embedded LaTeX to render. I don't really want to use Unicode like H→bb̅, which looks kinda ugly.

**Reply by DrAndiLowe** [reply]
Never mind, I figured it out:

First, use this:
https://qwinsi.github.io/tex2typst-webapp/

That told me that I had to type this:
```
Investigated the search potential for the Higgs boson via the $$H -> b macron(b)$$ decay channel
```

Through trial and error, I discovered that I needed to enclose the expression in double dollar signs.

**Reply by DrAndiLowe** [reply]
I am realising with horror that my 20-year-old definitions.tex file with all my math symbols will need to be retyped and converted to Typst. Even the \LaTeX macro is not defined!

**Reply by DrAndiLowe** [reply]
OK, we do a bit of hacking...

Do this:
```
rendercv new "Your Name" --create-typst-templates
```
From: [https://docs.rendercv.com/user_guide/how_to/override_default_templates/](https://docs.rendercv.com/user_guide/how_to/override_default_templates/)

Open Preamble.j2.typ, which is the only file I could fine that had
```
// Import the rendercv function and all the refactored components
#import "@preview/rendercv:0.1.0": *
```
Add this line at the top:
```
#import "@preview/metalogo:1.2.0": TeX, LaTeX
```

In your YAML, do:
```
You can use $$#LaTeX$$ now.
```

See also: [Implementing LaTeX macros unsupported by Typst
](https://qwinsi.github.io/tex2typst-webapp/impl-in-typst.html#the-latex-and-tex-macro)

**Comment by rahulw1981** [comment]
The site is not working. I am not able to edit my resume

**Reply by sinaatalay** [reply]
We apologize for the disruption. Due to a mistake on our side, our database is currently experiencing an outage. You can still render CVs, but saved CVs are temporarily unavailable.

Below is the message we are sharing on the app:

> Service Disruption Notice
> 
> Something went wrong on our backend, and there is currently an outage on our database. You may not be able to access your previously saved CVs. You can still render CVs, but you cannot access or create new saved ones for now.
> 
> We are working really hard to resolve this as soon as possible.
> 
> We know you put your trust in us, and we genuinely appreciate it. This was an honest mistake on our side, and we don't want to let you down. We have been providing this service for free for 2 years as a side project, and we have never worked on it full-time, so we kindly ask for your understanding.
> 
> — The RenderCV Team

**Comment by joeldacj** [comment]
The site (https://app.rendercv.com/) is unavailable, it says "Site Not Found" with an error from Firebase.

**Reply by sinaatalay** [reply]
Thank you for reporting. It's fixed now.
